File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        d = {}
        res = []
        for i in nums:
            d[i] = d.get(i, 0) + 1
        for i in range(k):
            res.append(max(d, key=d.get))
            del d[max(d, key=d.get)]
        return res

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.cpp
#include <unordered_map>
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char, int> hashMap;
        int l = 0;
        int res = 0;
        int maxFreq = 0;
        for (int r = 0; r < s.size(); ++r) {
            if (hashMap.contains(s[r])) {
                hashMap[s[r]] += 1;
            } else {
                hashMap[s[r]] = 1;
            }
            maxFreq = max(maxFreq, hashMap[s[r]]);
            
            while ((r - l) - maxFreq >= k) {
                --hashMap[s[l]];
                ++l;
            }
            res = max(res, r - l + 1);
        }
        return res;
    }
};
